Рефактор кода

nk2IsHere opened the pull request:
выделил пункты, которые необходимо исправлять и примерный план действий:
1)code review: наличие логики (кроме парсинга данных) в моделях нежелательно
2)code review: наличие взаимодействия с ui в логике может  привести к неожиданным и случайным последствиям
(Зачастую из-за гонки потоков и накладывающих запросов приводит к крашам и потере данных)

3)code review: желательно обрабатывать исключения с помощью сервисов по типу crashlytics
(Удобнее смотреть и исправлять баги на продакшне)
4)code review: наличие взаимодействия с ui здесь может привести к спагетти-коду
(Значительно понижает поддерживаемость и улучшаемость кодовой базы)
5)code review: наличие взаимодействия с ui в сервисе "гарантировано" приводит к крашам
6)code review: желательно интегрировать dependency injection в проект
(Уменьшает количество синглтонотов, показывает все обращения к общим компонентам)
7)code review: (all adapters) возможна замена groupie или аналогами
(Уменьшает количество boilerplate-кода в адаптерах)
8)code review: с activity желательно снять управление логикой и сетью, оставив управление UI 
(активити на данный момент - "объекты-боги", управляющие всем; необходимо четкое разграничение между данными, логикой и ui)

Примерный план моих действий:
0) Понять принцип, логику работы и фичи приложения
1) Ввести di, основанный на koin, общие компоненты интегрировать в него
2) Избавиться от логики в моделях, выделить репозитории и интеракторы(менеджеры);
перевести систему хранения локальных данных на objectbox (дб)
3) выделить в activitях взаимодействие с ui, сократить их количество
для этого буду использовать свою библиотеку, организующую систему событие - обработка - действие
[в этот момент начну писать документацию по работе с системой событий]
4) все взаимодействие с сетью и дб переведу на асинхронное с помощью rxjava
5) адаптеры перепишу на более простые аналоги с помощью groupie
6) сервисы (есть вопрос об их необходимости) адаптирую под взаимодействие с асинхронными потоками
7) приведу код в единый стиль, добывлю логгирование на внутреннюю память для дебаг-сборок и crashlytics для production

итого: примерно 10-14 дней работы
3.11 начало работы

про оплату:
если есть возможность оплатить по окончании работы - это будет лучшим вариантом
[прим. оценка - 7000р, если желаете, создам инд. предложение на платформе]
иначе, не буду против оплаты по часам
[125 р/час, 4 часа в день макс.]

nk2IsHere committed:
code review: models.kt

nk2IsHere committed:
code review: syncUtils

nk2IsHere committed:
code review: GlobalReceiver

nk2IsHere committed:
code review: APIProvider

nk2IsHere committed:
code review: App

nk2IsHere committed:
code review: adapters

Alexey-Ignatov commented:
Давайте, отличный анализ приложения, цена устраивает. 
Давайте сделаем этот рефакторинг, если все пойдет хорошо и мы пройдем ревью, то на следующие задачи увеличим ставку. Задач по этим кассам у нас много, можно будет это обсудить при случае.

Еще хотелось бы сказать, что нам важно как можно быстрее это все сделать. Потому что с июля мы не можем ничего опубликовать.

Есть один момент. это приложение для кассы Эвотор(https://evotor.ru/).
Это влияет на:
1) некоторые сервисы, которые, например, реагируют на печать чека
2) на структуру андроид манифеста, там нужно по-особому все указывать, чтобы это корректно работало на кассе
3) есть пара системных вызовов, но это по ним все видно.

Давайте приступать!
Вам, вероятно, интереснее было бы рассчитываться вне площадки, чтобы не платить там конскую комиссию. Я за этот вариант, тем более, что планы у нас далекие.
Чтобы вам было спокойнее, могу аванс закинуть, например, 3к.

nk2IsHere commented:
Хорошо, согласен, сам был очень неприятно удивлен таким комиссиям
Предлагаю перевести общение в телеграм (или подобный):
@nk2ishere

Alexey-Ignatov commented:
Не могу вас найти почему-то в телеграме.
Попробуйте вы меня:  +79533700495

Commit:
Created initial structure
Added nesessary libraries
Imported old pojos to new package

Commit:
Fixed models
Added token activity in new flavour

Commit:
Finished MasterConsole
Fixed db relations

Commit:
Fixed design flaw in recycler hooks

Commit:
UNTESTED!
Added order search
Added order pre/post pay creation
Added order good selection

Commit:
Added phone masks (untested)

Commit:
Added OrderFinal
Fixed OrderPay and GoodAdd
Fixed data

Commit:
Added services
Added current order handling
Added orderfinal activity wrapper
